engines/director/score.cpp

```diff
@@ -1344,123 +1357,273 @@ void Score::playQueuedSound() {
 void Score::loadFrames(Common::SeekableReadStreamEndian &stream, uint16 version) {
 	debugC(1, kDebugLoading, "****** Loading frames VWSC");
 
+	// Setup our streams for frames processing!
+	int dataSize = stream.size();
+	byte *data = (byte *)malloc(dataSize);
+	stream.read(data, dataSize);
+	
+	_framesStream = new Common::MemoryReadStreamEndian(data, dataSize, stream.isBE(), DisposeAfterUse::YES);
+
 	if (debugChannelSet(8, kDebugLoading)) {
-		stream.hexdump(stream.size());
+		_framesStream->hexdump(_framesStream->size());
 	}
 
-	uint32 size = stream.readUint32();
-	size -= 4;
+	_framesStreamSize = _framesStream->readUint32();
 
 	if (version < kFileVer400) {
 		_numChannelsDisplayed = 30;
 	} else if (version >= kFileVer400 && version < kFileVer600) {
-		uint32 frame1Offset = stream.readUint32();
-		uint32 numFrames = stream.readUint32();
-		uint16 framesVersion = stream.readUint16();
-		uint16 spriteRecordSize = stream.readUint16();
-		uint16 numChannels = stream.readUint16();
-		size -= 14;
-
-		if (framesVersion > 13) {
-			_numChannelsDisplayed = stream.readUint16();
+		uint32 frame1Offset = _framesStream->readUint32();
+		_numFrames = _framesStream->readUint32();
+		_framesVersion = _framesStream->readUint16();
+		uint16 spriteRecordSize = _framesStream->readUint16();
+		_numChannels = _framesStream->readUint16();
+
+		if (_framesVersion > 13) {
+			_numChannelsDisplayed = _framesStream->readUint16();
 		} else {
-			if (framesVersion <= 7)	// Director5
+			if (_framesVersion <= 7)    // Director5
 				_numChannelsDisplayed = 48;
 			else
-				_numChannelsDisplayed = 120;	// D6
+				_numChannelsDisplayed = 120;    // D6
 
-			stream.readUint16(); // Skip
+			_framesStream->readUint16(); // Skip
 		}
 
-		size -= 2;
-
 		warning("STUB: Score::loadFrames. frame1Offset: %x numFrames: %x version: %x spriteRecordSize: %x numChannels: %x numChannelsDisplayed: %x",
-			frame1Offset, numFrames, framesVersion, spriteRecordSize, numChannels, _numChannelsDisplayed);
+			frame1Offset, _numFrames, _framesVersion, spriteRecordSize, _numChannels, _numChannelsDisplayed);
 		// Unknown, some bytes - constant (refer to contuinity).
 	}
 
-	uint16 channelSize;
-	uint16 channelOffset;
+	// TODO How to work with channelData that gets overridden
+	// partically by channels, hence we keep it and read the score from left to right
+	// TODO Merge it with shared cast
+	memset(_channelData, 0, kChannelDataSize);
 
-	Frame *initial = new Frame(this, _numChannelsDisplayed);
-	// Push a frame at frame#0 position.
 	// This makes all indexing simpler
-	_frames.push_back(initial);
+	_frameOffsets.push_back(0);
+
+	_currentTempo = 0;
+	_currentPaletteId = CastMemberID(0, 0);
+
+	// Prepare frameOffsets
+	_version = version;
+	_frameOffsets.push_back(_framesStream->pos());
+
+	// In case number of frames not known, precompute them!
```

Comment:
I thought you needed to precompute it always because the header could be incorrect.